src/main/java/com/mongodb/jdbc/MongoConnection.java

```diff
@@ -223,26 +223,31 @@ public PreparedStatement prepareStatement(String sql) throws SQLException {
 
     @Override
     public void setAutoCommit(boolean autoCommit) throws SQLException {
+        // no-op, we only check that the connection is open
         checkConnection();
-        if (autoCommit) {
-            throw new SQLFeatureNotSupportedException(
-                    Thread.currentThread().getStackTrace()[1].toString());
-        }
+        logger.log(
+                Level.WARNING,
+                "Changing the auto-commit mode has no effect. The driver doesn't support transactions and is read-only."
```

Comment:
```suggestion
                "Changing the auto-commit mode has no effect. The driver doesn't support transactions and is read-only. "
```
nit: adding a space after period.